packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/mcp/formatters.py

```
@@ -194,6 +194,73 @@ def format_similar_documents(similar_docs: list[dict[str, Any]]) -> str:
+        # Create similarity index with minimal data
+        similarity_index = []
+        for doc_info in similar_docs:
+            document = doc_info.get("document", {})
+            
+            # Handle both HybridSearchResult objects and dict formats
+            if hasattr(document, 'document_id'):
+                doc_id = document.document_id
+                title = getattr(document, 'source_title', None) or "Untitled"
+                source_type = document.source_type
+                text_val = getattr(document, 'text', None)
+                text_length = len(text_val) if isinstance(text_val, str) else (int(text_val) if isinstance(text_val, (int, float)) else 0)
```

Ruff (0.12.2)

Use `X | Y` in `isinstance` call instead of `(X, Y)`

Convert to `X | Y`

(UP038)